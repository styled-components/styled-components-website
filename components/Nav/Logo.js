import styled, { css } from 'styled-components';

import rem from '../../utils/rem';

const Logo = styled.div`
  display: inline-block;
  vertical-align: middle;
  box-sizing: border-box;

  background-position: center;
  background-size: contain;

  ${p =>
    p.compact
      ? css`
          background-image: url(/icon.png);
          width: ${rem(100)};
          height: ${rem(30)};
        `
      : css`
          background-image: url(/nav-logo.png);
          width: ${rem(164)};
          height: ${rem(40)};
        `};
`;

export default Logo;
