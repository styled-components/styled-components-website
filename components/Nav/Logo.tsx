import styled from 'styled-components';
import rem from '../../utils/rem';

const Logo = styled.div`
  background-image: url(/nav-logo.png);
  background-position: center;
  background-size: contain;
  box-sizing: border-box;
  display: inline-block;
  height: ${rem(40)};
  width: ${rem(164)};
  vertical-align: middle;
`;

export default Logo;
