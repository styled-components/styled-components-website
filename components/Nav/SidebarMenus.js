import React, { useCallback, useState, useEffect } from 'react';
import styled from 'styled-components';
import { withRouter } from 'next/router';

import rem from '../../utils/rem';
import titleToDash from '../../utils/titleToDash';
import json from '../../pages/docs.json';
import Link, { StyledLink } from '../Link';

const { pages } = json;

const MenuInner = styled.div`
  display: block;
  box-sizing: border-box;
  height: 100%;
  padding-top: ${rem(60)};
`;

const TopLevelLink = styled.div`
  display: block;
  margin: ${rem(10)} ${rem(40)};
`;

const Section = styled.div`
  margin-bottom: ${rem(20)};
`;

const SectionTitle = styled.h4`
  display: block;
  margin: ${rem(10)} ${rem(40)};
  font-weight: normal;
`;

const SubSection = styled.h5`
  display: block;
  margin: ${rem(10)} ${rem(40)} ${rem(10)} ${rem(55)};
  font-size: 0.9rem;
  font-weight: normal;
`;

function Folder({ children, isOpenDefault = false, ...props }) {
  const [isOpen, setIsOpen] = useState(isOpenDefault);

  const toggleSubSections = useCallback(() => {
    setIsOpen(!isOpen);
  }, [isOpen]);

  useEffect(() => {
    if (isOpen !== isOpenDefault) setIsOpen(!!isOpenDefault);
    // it's fine to grab the current value of isOpen when isOpenDefault changes, if ever
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isOpenDefault]);

  return typeof children === 'function'
    ? children({
        rootProps: props,
        toggleSubSections,
        isOpen,
      })
    : null;
}

export const DocsSidebarMenu = withRouter(({ onRouteChange, router }) => (
  <MenuInner>
    {pages.map(({ title, pathname, sections }) => (
      <Folder key={title} isOpenDefault={router && router.pathname === `/docs/${pathname}`}>
        {({ rootProps, toggleSubSections, isOpen }) => (
          <Section {...rootProps} onClick={onRouteChange}>
            <SectionTitle onClick={toggleSubSections}>
              <Link href={`/docs/${pathname}`}>{title}</Link>
            </SectionTitle>

            {isOpen &&
              sections.map(({ title }) => (
                <SubSection key={title}>
                  <StyledLink href={`/docs/${pathname}#${titleToDash(title)}`}>{title}</StyledLink>
                </SubSection>
              ))}
          </Section>
        )}
      </Folder>
    ))}
  </MenuInner>
));

function getSectionPath(parentPathname, title) {
  return `${parentPathname || ''}#${titleToDash(title)}`;
}

function isFolderOpen(currentHref, { pathname, title, sections }) {
  return (
    sections.reduce((sum, v) => sum || window.location.href.endsWith(getSectionPath(pathname, v.title)), false) ||
    window.location.href.endsWith(pathname || '#' + titleToDash(title))
  );
}

export const SimpleSidebarMenu = ({ onRouteChange, pages = [] }) => (
  <MenuInner>
    {pages.map(({ title, pathname, sections, href }) => {
      if (!sections) {
        return (
          <TopLevelLink key={title}>
            <StyledLink href={pathname || '#' + (href || titleToDash(title))}>{title}</StyledLink>
          </TopLevelLink>
        );
      }

      return (
        <Folder
          key={title}
          isOpenDefault={
            typeof window !== 'undefined' && isFolderOpen(window.location.href, { title, pathname, sections })
          }
        >
          {({ rootProps, toggleSubSections, isOpen }) => (
            <Section {...rootProps} onClick={onRouteChange}>
              <SectionTitle onClick={toggleSubSections}>
                <Link href={pathname || '#' + titleToDash(title)}>{title}</Link>
              </SectionTitle>

              {isOpen &&
                sections.map(({ title }) => (
                  <SubSection key={title}>
                    <StyledLink unstyled href={getSectionPath(pathname, title)}>
                      {title}
                    </StyledLink>
                  </SubSection>
                ))}
            </Section>
          )}
        </Folder>
      );
    })}
  </MenuInner>
);
