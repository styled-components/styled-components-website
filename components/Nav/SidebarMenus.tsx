import { useRouter } from 'next/router';
import styled from 'styled-components';
import json from '../../pages/docs.json';
import rem from '../../utils/rem';
import titleToDash from '../../utils/titleToDash';
import Link, { StyledLink } from '../Link';

const { pages } = json;

export interface SimpleSidebarMenuProps {
  pages?: { title: string; pathname: string; sections: { title: string }[]; href: string }[];
}

export const SimpleSidebarMenu = ({ pages = [] }: SimpleSidebarMenuProps) => {
  return (
    <MenuInner>
      {pages.map(({ title, pathname, sections, href }, idx) => {
        if (!sections?.length) {
          return (
            <TopLevelLink key={idx}>
              <StyledLink href={pathname || '#' + (href || titleToDash(title))}>{title}</StyledLink>
            </TopLevelLink>
          );
        }

        return (
          <Section>
            <SectionTitle>
              <Link href={pathname || '#' + titleToDash(title)}>{title}</Link>
            </SectionTitle>

            <SubSection key={title}>
              <StyledLink href={getSectionPath(pathname, title)}>{title}</StyledLink>
            </SubSection>
          </Section>
        );
      })}
    </MenuInner>
  );
};

const MenuInner = styled.div`
  display: block;
  box-sizing: border-box;
  height: 100%;
  padding-top: ${rem(60)};
`;

const TopLevelLink = styled.div`
  display: block;
  margin: ${rem(10)} ${rem(40)};
`;

const Section = styled.div`
  margin-bottom: ${rem(20)};
`;

const SectionTitle = styled.h4`
  display: block;
  margin: ${rem(10)} ${rem(40)};
`;

const SubSection = styled.h5`
  display: block;
  margin: ${rem(10)} ${rem(40)} ${rem(10)} ${rem(55)};
  font-size: 0.9rem;
  font-weight: normal;
`;

export interface DocsSidebarMenuProps {}

export const DocsSidebarMenu = (props: DocsSidebarMenuProps) => {
  const router = useRouter();

  return (
    <MenuInner>
      {pages.map(({ title, pathname, sections }) => (
        <Section>
          <SectionTitle>
            <Link href={`/docs/${pathname}`}>{title}</Link>
          </SectionTitle>

          {sections.map(({ title }) => (
            <SubSection key={title}>
              <StyledLink href={`/docs/${pathname}#${titleToDash(title)}`}>{title}</StyledLink>
            </SubSection>
          ))}
        </Section>
      ))}
    </MenuInner>
  );
};

function getSectionPath(parentPathname: string, title: string) {
  return `${parentPathname}#${titleToDash(title)}`;
}
