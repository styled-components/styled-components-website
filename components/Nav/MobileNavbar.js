import React from 'react';
import styled, { css } from 'styled-components';
import { Search, KeyboardArrowDown } from '@styled-icons/material';
import rem from '../../utils/rem';
import { navbarHeight } from '../../utils/sizes';
import { paleGrey } from '../../utils/colors';
import { mobile } from '../../utils/media';
import { CloseIcon, FoldIcon } from './NavIcons';
import Link from '../Link';
import NavLinks from './NavLinks';
import Social from './Social';
import Logo from './Logo';
import NavButton from './NavButton';

const Wrapper = styled.div`
  display: none;

  ${mobile(css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: ${rem(navbarHeight)};
  `)};
`;

const SecondaryMenu = styled.div`
  position: absolute;
  top: ${rem(navbarHeight)};
  left: 0;
  right: 0;

  ${p =>
    p.isOpen
      ? css`
          height: ${rem(navbarHeight)};
        `
      : css`
          height: 0;
        `} display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  padding: 0 ${rem(20)};
  transition: height 0.1s;

  user-select: none;
  -webkit-overflow-scrolling: touch;
  overflow-x: scroll;
  overflow-y: hidden;

  background: ${paleGrey};
  color: #868686;
`;

const LogoLink = styled(Link).attrs((/* props */) => ({
  unstyled: true,
  href: '/',
  'aria-label': 'styled components',
}))`
  display: inline-block;
  vertical-align: center;
`;

const ArrowWrapper = styled.div`
  transition: transform 0.2s;

  ${p =>
    p.shouldRotate &&
    css`
      transform-origin: center center;
      transform: rotate(180deg);
    `};
`;

const SecondaryMenuItem = styled.div`
  padding-right: ${rem(20)};
`;

const StyledIcon = styled.div`
  && {
    width: ${p => rem(p.size || 20)};
    height: ${p => rem(p.size || 20)};
  }
`;

const MobileNavbar = props => {
  const { isSideFolded, isMobileNavFolded, onSideToggle, onMobileNavToggle, showSideNav, onSearchButtonClick } = props;

  return (
    <Wrapper>
      {showSideNav !== false && (
        <NavButton active={!isSideFolded} onClick={onSideToggle}>
          {isSideFolded ? <FoldIcon /> : <CloseIcon />}
        </NavButton>
      )}

      <LogoLink>
        <Logo compact />
      </LogoLink>
      <div>
        <NavButton onClick={onSearchButtonClick}>
          <StyledIcon as={Search} size={28} />
        </NavButton>

        <NavButton onClick={onMobileNavToggle} active={!isMobileNavFolded}>
          <ArrowWrapper shouldRotate={!isMobileNavFolded}>
            <StyledIcon as={KeyboardArrowDown} size={36} />
          </ArrowWrapper>
        </NavButton>
      </div>

      <SecondaryMenu isOpen={!isMobileNavFolded}>
        <NavLinks />
        <SecondaryMenuItem>
          <Social />
        </SecondaryMenuItem>
      </SecondaryMenu>
    </Wrapper>
  );
};

export default MobileNavbar;
