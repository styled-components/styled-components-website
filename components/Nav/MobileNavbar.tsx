import { KeyboardArrowDown } from '@styled-icons/material';
import styled, { css } from 'styled-components';
import { paleGrey } from '../../utils/colors';
import { mobile } from '../../utils/media';
import rem from '../../utils/rem';
import { navbarHeight } from '../../utils/sizes';
import Link from '../Link';
import Logo from './Logo';
import NavButton from './NavButton';
import { CloseIcon, FoldIcon } from './NavIcons';
import NavLinks from './NavLinks';
import Social from './Social';

const Wrapper = styled.div`
  display: none;

  ${mobile(css`
    align-items: center;
    display: flex;
    flex-grow: 1;
    height: ${rem(navbarHeight)};
    justify-content: space-between;
  `)};
`;

const SecondaryMenu = styled.div<{ $isOpen?: boolean }>`
  position: absolute;
  top: ${rem(navbarHeight)};
  left: 0;
  right: 0;

  ${p =>
    p.$isOpen
      ? css`
          height: ${rem(navbarHeight)};
        `
      : css`
          height: 0;
        `} display: flex;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-between;
  padding: 0 ${rem(20)};
  transition: height 0.1s;

  user-select: none;
  -webkit-overflow-scrolling: touch;
  overflow-x: scroll;
  overflow-y: hidden;

  background: ${paleGrey};
  color: #868686;
`;

const LogoLink = styled(Link).attrs((/* props */) => ({
  unstyled: true,
  href: '/',
  'aria-label': 'styled components',
}))`
  display: inline-block;
  vertical-align: center;
`;

const ArrowWrapper = styled.div<{ $shouldRotate?: boolean }>`
  transition: transform 0.2s;

  ${p =>
    p.$shouldRotate &&
    css`
      transform-origin: center center;
      transform: rotate(180deg);
    `};
`;

const SecondaryMenuItem = styled.div`
  padding-right: ${rem(20)};
`;

const StyledIcon = styled.div<{ $size?: number }>`
  && {
    width: ${p => rem(p.$size || 20)};
    height: ${p => rem(p.$size || 20)};
  }
`;

export interface MobileNavbarProps {
  isMobileNavFolded?: boolean;
  isSideFolded?: boolean;
  onMobileNavToggle?: () => void;
  onSideToggle?: () => void;
  showSideNav?: boolean;
}

const MobileNavbar = ({
  children,
  isMobileNavFolded,
  isSideFolded,
  onMobileNavToggle,
  onSideToggle,
  showSideNav,
}: React.PropsWithChildren<MobileNavbarProps>) => {
  return (
    <Wrapper>
      {showSideNav !== false && (
        <NavButton $active={!isSideFolded} onClick={onSideToggle}>
          {isSideFolded ? <FoldIcon /> : <CloseIcon />}
        </NavButton>
      )}

      <LogoLink>
        <Logo $compact />
      </LogoLink>

      {children}

      <NavButton onClick={onMobileNavToggle} $active={!isMobileNavFolded} style={{ position: 'absolute', right: 0 }}>
        <ArrowWrapper $shouldRotate={!isMobileNavFolded}>
          <StyledIcon as={KeyboardArrowDown} $size={36} />
        </ArrowWrapper>
      </NavButton>

      <SecondaryMenu $isOpen={!isMobileNavFolded}>
        <NavLinks />
        <SecondaryMenuItem>
          <Social />
        </SecondaryMenuItem>
      </SecondaryMenu>
    </Wrapper>
  );
};

export default MobileNavbar;
