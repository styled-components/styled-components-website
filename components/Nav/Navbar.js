import React, { PureComponent, createRef } from 'react';
import styled, { css } from 'styled-components';
import { Close } from '@styled-icons/material';
import rem from '../../utils/rem';
import { blmGrey, paleGrey } from '../../utils/colors';
import { navbarHeight } from '../../utils/sizes';
import { headerFont } from '../../utils/fonts';
import { mobile } from '../../utils/media';
import Link from '../Link';
import NavLinks from './NavLinks';
import Social from './Social';
import Logo from './Logo';
import MobileNavbar from './MobileNavbar';
import SearchWithAlgolia from './SearchWithAlgolia';

const Wrapper = styled.nav`
  position: fixed;
  left: 0;
  box-sizing: border-box;
  z-index: 3;

  width: 100%;
  height: ${rem(navbarHeight)};

  font-family: ${headerFont};
  font-size: ${rem(15)};
  font-weight: 500;
  background: ${(props) => (props.transparent ? 'transparent' : blmGrey)};
  transition: background 300ms ease-out;
  color: white;
  padding: 0;
`;

const StartWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
`;

const EndWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
`;
/* stylelint-disable */
const StyledSocial = styled(Social)``;
/* stylelint-enable */

const NormalNavbar = styled.div`
  display: flex;
  align-items: center;
  padding: 0 ${rem(20)};
  justify-content: space-between;
  ${StartWrapper}, ${EndWrapper} ${StyledSocial} {
    ${mobile(css`
      display: none;
    `)};
  }
`;

const StyledModalCloseIcon = styled(Close)`
  width: ${rem(26)};
  height: ${rem(26)};
  color: white;
`;

const AlgoliaModal = styled.div`
  ${mobile(css`
    background: currentColor;
    overflow: auto;
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
  `)};
`;

const baseZ = 1;

const AlgoliaModalHeader = styled.div`
  display: none;
  color: currentColor;

  ${mobile(css`
    display: ${(props) => (props.isOpen ? 'block' : 'none')};

    button {
      cursor: pointer;
      padding: 0;
      position: fixed;
      right: ${rem(10)};
      top: ${rem(11)};
      border: none;
      z-index: ${baseZ + 1};
    }
  `)};
`;

const AlgoliaModalOverlay = styled.div`
  margin-right: ${rem(10)};

  ${mobile(css`
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: ${baseZ};
    left: 0;
    display: ${(props) => (props.isOpen ? 'block' : 'none')};
    background: ${paleGrey};
    overflow-y: auto;
    margin: 0;

    .algolia-autocomplete .ds-dropdown-menu {
      position: static !important;
      display: block;
      max-width: 100%;
      min-width: 0;
    }
  `)};
`;

class ModalContainer extends PureComponent {
  modalElement = createRef();
  onModalOverlayClick = (e) => {
    if (!this.modalElement.current.contains(e.target)) {
      this.props.requestModalClose();
    }
  };
  onCloseButtonClick = (e) => {
    e.stopPropagation();
    this.props.requestModalClose();
  };
  render() {
    return (
      <>
        <AlgoliaModalHeader isOpen={this.props.isOpen}>
          <button onClick={this.onCloseButtonClick}>
            <StyledModalCloseIcon />
          </button>
        </AlgoliaModalHeader>
        <AlgoliaModalOverlay onClick={this.onModalOverlayClick} isOpen={this.props.isOpen}>
          <AlgoliaModal ref={this.modalElement}>
            <div>{this.props.children}</div>
          </AlgoliaModal>
        </AlgoliaModalOverlay>
      </>
    );
  }
}

const LogoLink = styled(Link).attrs((/* props */) => ({
  unstyled: true,
  href: '/',
}))`
  display: inline-block;
  vertical-align: center;
  margin-right: ${rem(35)};
`;

class Navbar extends PureComponent {
  state = {
    isOpen: false,
  };
  openModal = () => this.setState(() => ({ isOpen: true }));
  closeModal = () => this.setState(({ isOpen }) => (isOpen ? { isOpen: false } : null));
  render() {
    const { onSideToggle, onMobileNavToggle, isSideFolded, isMobileNavFolded, showSideNav, transparent } = this.props;

    return (
      <Wrapper transparent={transparent}>
        <NormalNavbar>
          <StartWrapper>
            <LogoLink>
              <Logo />
            </LogoLink>
            <NavLinks />
          </StartWrapper>
          <EndWrapper>
            <ModalContainer isOpen={this.state.isOpen} requestModalClose={this.closeModal}>
              <SearchWithAlgolia requestModalClose={this.closeModal} />
            </ModalContainer>
            <StyledSocial />
          </EndWrapper>
        </NormalNavbar>
        <MobileNavbar
          isSideFolded={isSideFolded}
          onSearchButtonClick={this.openModal}
          isMobileNavFolded={isMobileNavFolded}
          onSideToggle={onSideToggle}
          onMobileNavToggle={onMobileNavToggle}
          showSideNav={showSideNav}
        />
      </Wrapper>
    );
  }
}

export default Navbar;
