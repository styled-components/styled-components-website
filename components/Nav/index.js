import React from 'react';
import Navbar from './Navbar';
import Sidebar from './Sidebar';
import { DocsSidebarMenu, SimpleSidebarMenu } from './SidebarMenus';

const Nav = props => {
  const {
    isSideFolded,
    isMobileNavFolded,
    onSideToggle,
    onMobileNavToggle,
    onRouteChange,
    showSideNav,
    useDocsSidebarMenu,
    transparent,
    pages,
  } = props;

  return (
    <div>
      <Navbar
        showSideNav={showSideNav}
        transparent={transparent}
        isSideFolded={isSideFolded}
        isMobileNavFolded={isMobileNavFolded}
        onSideToggle={onSideToggle}
        onMobileNavToggle={onMobileNavToggle}
      />

      {showSideNav !== false && (
        <Sidebar isFolded={isSideFolded}>
          {useDocsSidebarMenu !== false ? (
            <DocsSidebarMenu onRouteChange={onRouteChange} />
          ) : (
            <SimpleSidebarMenu onRouteChange={onRouteChange} pages={pages} />
          )}
        </Sidebar>
      )}
    </div>
  );
};

export default Nav;
