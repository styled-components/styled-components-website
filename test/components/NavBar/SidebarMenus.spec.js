import React from 'react';
import renderer from 'react-test-renderer';
import { DocsSidebarMenu, SimpleSidebarMenu } from '../../../components/Nav/SidebarMenus';

test('DocsSidebarMenu renders correctly', () => {
  const tree = renderer.create(<DocsSidebarMenu />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('SimpleSidebarMenu renders correctly', () => {
  const tree = renderer.create(<SimpleSidebarMenu />).toJSON();

  expect(tree).toMatchSnapshot();
});
