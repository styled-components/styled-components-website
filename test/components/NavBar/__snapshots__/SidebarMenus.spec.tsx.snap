// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocsSidebarMenu renders correctly 1`] = `
.c3 {
  display: inline-block;
  color: inherit;
  cursor: pointer;
  padding: 0.1111111111111111rem 0.4444444444444444rem;
  margin: -0.1111111111111111rem -0.4444444444444444rem;
}

.c0 {
  display: block;
  box-sizing: border-box;
  height: 100%;
  padding-top: 3.3333333333333335rem;
}

.c1 {
  margin-bottom: 1.1111111111111112rem;
}

.c2 {
  display: block;
  margin: 0.5555555555555556rem 2.2222222222222223rem;
}

.c4 {
  display: block;
  margin: 0.5555555555555556rem 2.2222222222222223rem 0.5555555555555556rem 3.0555555555555554rem;
  font-size: 0.9rem;
  font-weight: normal;
}

@media (min-width: 62.5em) {
  .c3 {
    border-radius: 0.16666666666666666rem;
  }

  .c3:hover {
    background: rgba(20, 20, 20, 0.1);
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h4
      className="c2"
    >
      <a
        className="c3"
      >
        Basics
      </a>
    </h4>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/basics#motivation"
      >
        Motivation
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/basics#installation"
      >
        Installation
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/basics#getting-started"
      >
        Getting Started
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/basics#adapting-based-on-props"
      >
        Adapting based on props
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/basics#extending-styles"
      >
        Extending styles
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/basics#styling-any-component"
      >
        Styling any component
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/basics#passed-props"
      >
        Passed props
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/basics#coming-from-css"
      >
        Coming from CSS
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/basics#attaching-additional-props"
      >
        Attaching additional props
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/basics#animations"
      >
        Animations
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/basics#react-native"
      >
        React Native
      </a>
    </h5>
  </div>
  <div
    className="c1"
  >
    <h4
      className="c2"
    >
      <a
        className="c3"
      >
        Advanced
      </a>
    </h4>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/advanced#theming"
      >
        Theming
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/advanced#refs"
      >
        Refs
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/advanced#security"
      >
        Security
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/advanced#existing-css"
      >
        Existing CSS
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/advanced#tagged-template-literals"
      >
        Tagged Template Literals
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/advanced#server-side-rendering"
      >
        Server Side Rendering
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/advanced#referring-to-other-components"
      >
        Referring to other components
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/advanced#style-objects"
      >
        Style Objects
      </a>
    </h5>
  </div>
  <div
    className="c1"
  >
    <h4
      className="c2"
    >
      <a
        className="c3"
      >
        API Reference
      </a>
    </h4>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/api#primary"
      >
        Primary
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/api#helpers"
      >
        Helpers
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/api#test-utilities"
      >
        Test Utilities
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/api#supported-css"
      >
        Supported CSS
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/api#typescript"
      >
        TypeScript
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/api#previous-apis"
      >
        Previous APIs
      </a>
    </h5>
  </div>
  <div
    className="c1"
  >
    <h4
      className="c2"
    >
      <a
        className="c3"
      >
        Tooling
      </a>
    </h4>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/tooling#babel-plugin"
      >
        Babel Plugin
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/tooling#babel-macro"
      >
        Babel Macro
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/tooling#typescript-plugin"
      >
        TypeScript Plugin
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/tooling#jest-integration"
      >
        Jest Integration
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/tooling#stylelint"
      >
        Stylelint
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/tooling#styled-theming"
      >
        Styled Theming
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/tooling#syntax-highlighting"
      >
        Syntax Highlighting
      </a>
    </h5>
  </div>
  <div
    className="c1"
  >
    <h4
      className="c2"
    >
      <a
        className="c3"
      >
        FAQs
      </a>
    </h4>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#what-do-i-need-to-do-to-migrate-to-v6"
      >
        What do I need to do to migrate to v6?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#what-do-i-need-to-do-to-migrate-to-v5"
      >
        What do I need to do to migrate to v5?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#what-do-i-need-to-do-to-migrate-to-v4"
      >
        What do I need to do to migrate to v4?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#can-i-nest-rules"
      >
        Can I nest rules?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#can-i-use-css-frameworks"
      >
        Can I use css frameworks?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#how-can-i-override-styles-with-higher-specificity"
      >
        How can I override styles with higher specificity?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#how-can-i-override-inline-styles"
      >
        How can I override inline styles?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#why-do-my-dom-nodes-have-two-classes"
      >
        Why do my DOM nodes have two classes?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#when-to-use-attrs"
      >
        When to use attrs?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#why-should-i-avoid-declaring-styled-components-in-the-render-method"
      >
        Why should I avoid declaring styled components in the render method?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#i-am-a-library-author.-should-i-bundle-styled-components-with-my-library"
      >
        I am a library author. Should I bundle styled-components with my library?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#why-am-i-getting-a-warning-about-several-instances-of-module-on-the-page"
      >
        Why am I getting a warning about several instances of module on the page?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#why-am-i-getting-html-attribute-warnings"
      >
        Why am I getting HTML attribute warnings?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#which-browsers-are-supported"
      >
        Which browsers are supported?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#how-do-i-use-styled-components-with-create-react-app"
      >
        How do I use styled-components with create-react-app?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#how-can-i-fix-issues-when-using-npm-link-or-yarn-link"
      >
        How can I fix issues when using npm link or yarn link?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#how-do-i-fix-flickering-text-after-server-side-rendering"
      >
        How do I fix flickering text after server side rendering?
      </a>
    </h5>
    <h5
      className="c4"
    >
      <a
        className="c3"
        href="/docs/faqs#missing-declarations-for-styled-componentsnative"
      >
        Missing Declarations for styled-components/native?
      </a>
    </h5>
  </div>
</div>
`;

exports[`SimpleSidebarMenu renders correctly 1`] = `
.c0 {
  display: block;
  box-sizing: border-box;
  height: 100%;
  padding-top: 3.3333333333333335rem;
}

<div
  className="c0"
/>
`;
